app/controllers/two_factor_authentication/otp_expired_controller.rb

```diff
@@ -0,0 +1,25 @@
+module TwoFactorAuthentication
+  class OtpExpiredController < ApplicationController
+    before_action :validate_otp_expiration
+
+    def show
+      @otp_delivery_preference = otp_delivery_preference
+    end
+
+    private
+
+    def otp_delivery_preference
+      current_user.otp_delivery_preference
+    end
+
+    def validate_otp_expiration
+      redirect_to after_sign_in_path_for(current_user) if user_fully_authenticated? && !otp_expired?
```

Comment:
We will show this screen for users who are not "fully" authenticated, correct? (As in, they are in the process of completing the MFA step). I think if we want to redirect for those users, we'd want to change the condition to allow the not-fully-signed-in user.

```suggestion
      redirect_to after_sign_in_path_for(current_user) if user_signed_in? && !otp_expired?
```